### ISSUE TITLE
Duplicate projects

### ISSUE THREAD

**Issue by jakirkham** [appveyor/ci#1107]
I'm not entirely sure how we got into this situation, but it seems we have a duplicate project. Are these actually duplicates? Are there more of them? Is there a way to clean them up efficiently?

<img width="1554" alt="screen shot 2016-10-11 at 23 49 05" src="https://cloud.githubusercontent.com/assets/3019665/19296865/6dffe8ce-900d-11e6-9cb3-7a4678d4bdea.png">

**Comment by FeodorFitsner** [comment]
Just delete one of them.

**Comment by IlyaFinkelshteyn** [comment]
Note that probably you want to delete recently added one, which has less clear name if you look at browser address. It will be `<project_name>-<random_string>.

**Comment by jakirkham** [comment]
That's what you would think, right?

The other day I tried to stop a duplicate job (which is how this became apparent) and they both failed. Then I restarted one and they both restarted. Decided to wait it out and now I have this situation.

<img width="1554" alt="screen shot 2016-10-12 at 23 22 39" src="https://cloud.githubusercontent.com/assets/3019665/19335640/dabfdb66-90d2-11e6-995a-fcbe20ea5c68.png">

If I click on either project, it seems to link back to the exact same passing project with the same URL. If I click on the build of the failing project, I get this.

<img width="1554" alt="screen shot 2016-10-12 at 23 25 24" src="https://cloud.githubusercontent.com/assets/3019665/19335679/2b1cadc8-90d3-11e6-8097-e6ad1ef99a3a.png">

Though now it occurs to me that there might be duplicate webhooks, which there are. Not sure how this factors into the duplicate projects yet. For example, which webhook do I remove and if I remove one does it actually remove one of the projects too or is there more required?

**Comment by IlyaFinkelshteyn** [comment]
Hi,

It looks like with your automation you helped us to expose concurrency issue. We are working on it. 

For now, we fixed you specific issue manually by adding random string to the duplicate project's slug. You can safely go to https://ci.appveyor.com/project/conda-forge/importlib-feedstock-ksdwd and delete duplicate project.

Thank you,
Ilya.

**Comment by jakirkham** [comment]
Glad it helped. Thanks for changing the string. Was able to remove it and the correct webhook.

Would you like to keep this open for the concurrency issue or would you rather it be closed?

**Comment by IlyaFinkelshteyn** [comment]
Good, lets keep it open for now

**Comment by jakirkham** [comment]
Ran into another one with the [npctypes-feedstock]( https://github.com/conda-forge/npctypes-feedstock ). Any chance I could get help renaming one of these? Is there some way I could do this myself so as to save you the effort?

<img width="1554" alt="screen shot 2016-10-19 at 2 42 50" src="https://cloud.githubusercontent.com/assets/3019665/19508184/c2200040-95a5-11e6-9134-b8d0b4be841f.png">

**Comment by IlyaFinkelshteyn** [comment]
Yes, renamed the one which was created later to npctypes-feedstock-iksadq

Sorry for the trouble.

--ilya.

**Comment by jakirkham** [comment]
No worries. Thanks for helping us correct these.

**Comment by jakirkham** [comment]
Have another one for you [kenjutsu-feedstock]( https://ci.appveyor.com/projects?q=kenjutsu-feedstock ).

**Comment by IlyaFinkelshteyn** [comment]
OK, later created one renamed to kenjutsu-feedstock-lihaws :)

**Comment by jakirkham** [comment]
Thanks. Also just now found [ase-feedstock]( https://github.com/conda-forge/ase-feedstock ). This one was recently created if that helps at all.

**Comment by jakirkham** [comment]
Also [`lektor-feedstock`]( https://github.com/conda-forge/lektor-feedstock ) has two.

**Comment by jakirkham** [comment]
Seem to have figured out how to fixes these on my end. So I'll try to do the tidying up on my end so you aren't having to chase these down. 😄

**Comment by IlyaFinkelshteyn** [comment]
I have some more job for you :)

bob.ip.color-feedstock
defusedxml-feedstock
lazy-object-proxy-feedstock
python-slugify-feedstock
warc-feedstock

By the way, can you share how you do this cleanup?

--ilya.

**Comment by jakirkham** [comment]
Thanks for the list.

So I deleted the other `python-slugify` (as it had just built). Could you please rename the remaining one back to `python-slugify`?

**Comment by jakirkham** [comment]
`bob.ip.color` interestingly never should have been building. However, it seems the CI was building as recently as 9 days ago. I think one of the projects wasn't configured correctly when it got created for some reason. Not sure if that helps at all.

**Comment by jakirkham** [comment]
As far as how I do cleanup, I tried a few different things. However, it seems like I can get away with just deleting one of the projects from the web interface. Now it always picks the most recently built project it seems. Still it provides me a way to get rid of the extra one.

**Comment by jakirkham** [comment]
Have addressed all of the ones that you provided. Thanks again. 😄

**Comment by jakirkham** [comment]
Could someone please rename https://ci.appveyor.com/project/conda-forge/ruamel-yaml-feedstock-d73ox to https://ci.appveyor.com/project/conda-forge/ruamel-yaml-feedstock .

**Comment by IlyaFinkelshteyn** [comment]
We already have project with slug ruamel-yaml-feedstock connected to repository jakirkham/ruamel_yaml-feedstock (ruamel-yaml-feedstock-d73ox is connected to conda-forge/ruamel_yaml-feedstock).

**Comment by jakirkham** [comment]
Feel free to rename my fork's CI project name if need be.

**Comment by IlyaFinkelshteyn** [comment]
Done, fork's project slug is `ruamel-yaml-feedstock-sa7jk`

**Comment by jakirkham** [comment]
So we ran into this problem again recently. Appears [`conda-forge/keras-applications-feedstock`]( https://github.com/conda-forge/keras-applications-feedstock ) got added twice.

**Comment by FeodorFitsner** [comment]
Could you drop the links to those two projects?

**Comment by jakirkham** [comment]
Sure. Here you go.

* https://ci.appveyor.com/project/conda-forge/keras-applications-feedstock
* https://ci.appveyor.com/project/conda-forge/keras-applications-feedstock-na16w

**Comment by jakirkham** [comment]
So the issue (at least from our perspective) is that the same repo is getting added multiple times under the org. It seemed the issue had subsided for a while (or at least was less obvious to us). Not sure if something has changed with how repos are added that is allowing them to create multiple projects again. Would be nice to make sure that if a repo already exists within an org, it doesn't get added again as a new project.

**Comment by IlyaFinkelshteyn** [comment]
This was always allowed. A lot of customers create multiple projects per repo when they cannot fit different scenarios into single project. Good example is scheduled build project which runs full set of tests and on-demand builds project which runs only smoke tests.

In your project creation script, you can get [list of projects](https://www.appveyor.com/docs/api/projects-builds/#get-projects) and check if project with slug the same as repo name exists, and if yes, cancel project creation. Sure all projects with slug `<repo_name-random_string>` should be deleted before once manually.
